src/glue-layer/identifiers.ts

```diff
@@ -0,0 +1,6 @@
+export enum MUTATION_TYPES {
```

Comment:
You have to explain me the whole purpose of the glue layer. Anyways, I don't think that we should call it this way.

I'd like to make a diagram of the layers (I think you kind of did one in your proposal). It would be great to improve it and rationalize it.